### PR TITLE
Human readable time of day option

### PR DESCRIPTION
**Pull request by bierpub**
Create a Human Readable option for `Time of Day` that does not include the date that the rule started. This can be used as an alternate to the start date when wanting to print only the time of day for each occurrence and without the word `starting`.

Occurrences that are once per day will print the time of day
`monthly on the 2nd of the month at 4:15 PM`
Occurrences that are more than once per day will show the starting at time
`every 7 hours starting at 6:30 AM`

### CODE REVIEW

**Comment by rlanvin** [comment]
Hello! Thanks for the PR. It seems some tests break, could you have a look?

File: tests/RRuleTest.php [hunk 1]
```diff
@@ -3191,6 +3191,18 @@ public function humanReadableStrings()
 				"daily",
 				"daily"
 			),
+			array(
+				"DTSTART=20170202T161500Z\nRRULE:FREQ=MONTHLY;BYMONTHDAY=2",
```
Comment (rlanvin) [inline tests/RRuleTest.php:3195]: Tests are still breaking, probably because `DTSTART` takes `:` instead of `=`

File: src/i18n/de.php [hunk 2]
```diff
@@ -44,6 +44,8 @@
 		'else' => 'Alle %{interval} Sekunden'
 	),
 	'dtstart' => ', ab dem %{date}',
+	'timeofday' => ' um %{timeofday}',
+	'startingtimeofday' => ' starting at %{timeofday}',
```
Comment (rlanvin) [inline src/i18n/de.php:48]: that would be "ab %{timeofday}" in German

File: src/RRule.php [hunk 3]
```diff
@@ -2062,17 +2062,19 @@ static protected function i18nLoad($locale, $fallback = null, $use_intl = null,
...
+	 * | `locale`              | string  | The locale to use (autodetect)
+	 * | `fallback`            | string  | Fallback locale if main locale is not found (default en)
+	 * | `date_formatter`      | callable| Function used to format the date (takes date, returns formatted)
+	 * | `timeofday_formatter` | callable| Function used to format the timeofday (takes date, returns formatted)
+	 * | `explicit_inifite`    | bool    | Mention "forever" if the rule is infinite (true)
+	 * | `dtstart`             | bool    | Mention the start date (true)
+	 * | `include_start`       | bool    |
+	 * | `include_timeofday`   | bool    |
+	 * | `include_until`       | bool    |
+	 * | `custom_path`         | string  |
```
Comment (rlanvin) [inline src/RRule.php:2077]: Why are all the lines in the diff? Only the last one changed - is this an indentation change or something?

**Comment by rlanvin** [comment]
I left a few comments on the code, but I'd like to discuss the design of this feature. I see two main flaws in the current PR:

1. I don't like that enabling this new `include_timeofday` option forcibly disables `include_start`. It's a sign that logically this isn't a separate, independent option, but rather a variation of `include_start`. Therefore my first instinct is that it would make more sense to accept a 3rd value for `include_start` which would indicate that we want "time only" start (this could be a string value, or a constant). To preserve backward compatibility, `true` and `false` still need to work of course.
2. The new `timeofday_formatter` duplicates a lot of code from `date_formatter`, and because the two options are mutually exclusive (point 1), it's either one or the other, we'll never need both at the same time. So the duplication doesn't make sense, and I think `date_formatter` should be used instead of introducing a duplicated option.

Hope that makes sense. Let me know what you think.

**Comment by rlanvin** [comment]
Think about point 1 further, another approach could be to design this as a "sub option", similar to what `explicit_infinite` does (it only has an effect if `include_until` is set). It could be something like `start_time_only` true/false for example, that has no effect unless `include_start` is set. This would probably be even better as it would be more consistent with the rest of the api.

File: src/RRule.php (hunk 3 above)
Comment (bierpub) [reply src/RRule.php:2077]: Yes this is only because the new option names are longer so the table ended up wider.

File: tests/RRuleTest.php (hunk 1 above)
Comment (bierpub) [reply tests/RRuleTest.php]: Sorry. I think I got it straight now.

**Comment by bierpub** [comment]
I do agree there is some duplicate code now, and I was a little hesitant doing that. Having to properly update handle multiple languages is why I ended up down this route. But I'll look back at this and see if I can make this better. I like your suggestions.

**Comment by bierpub** [comment]
Hello. I think this is much better now. There is only 1 new option `start_time_only` to tell it to print the time with a slightly different string. For the time formatting, it is now using the `date_formatter` to set the time without the date if necessary.

**Comment by bierpub** [comment]
@rlanvin Hi. Any chance we could get this reviewed and merged into the master code base? I've been using this for a few months now in our production environment and all has been good with this. Thanks.

**Comment by rlanvin** [comment]
@bierpub Yes sure - sorry I thought I already reviewed and merged this one, but I'm bad at keeping track of the open PRs at the moment... Thanks a lot for addressing my comments from January. This looks good to me.

**Pull request merged**